src/layers/domain/response/validation_errors.py

```diff
@@ -70,14 +77,60 @@ def decorator(*args: P.args, **kwargs: P.kwargs):
     return decorator
 
 
+class DeviceType(str, Enum):
+    PRODUCT = "PRODUCT"
+    ENDPOINT = "ENDPOINT"
+
+    @classmethod
+    def _missing_(cls, value):
+        for member in cls:
+            if value.upper() == member.value:
+                return member
+
+
+class SearchQueryParams(BaseModel, extra=Extra.forbid):
+    device_type: DeviceType
```

Comment:
1. I think it would be good to import DeviceType from src/layers/domain/core/device.py instead
2. Can move the case-insensitivity to

```
class SearchQueryParams(BaseModel, extra=Extra.forbid):
    device_type: DeviceType

    @validator("device_type")
    def validate_device_type(device_type: str):
        return device_type.lower()

3. Would move these to a new module `layers/domain/api/query.py`